pkg/controller/upgrades.go

```diff
@@ -332,7 +332,12 @@ func (c *Controller) upgrade500(ctx context.Context, tenant *miniov2.Tenant) (*m
 	if err != nil && !k8serrors.IsNotFound(err) {
 		return tenant, err
 	}
-	if logSearchSecret != nil {
+	// Update the secret only if it exists
+	// logSearchSecret will not be nil even if secret is not found so use k8serrors.IsNotFound() to determine if secret was found or not
+	if k8serrors.IsNotFound(err) {
+		klog.Infof("Log Search Secret was not found, then it will not be updated")
```

Comment:
first of all we don't have to print what we don't have, we should just ignore and move on.